### ISSUE TITLE
Ability to select multiple entries and edit

### ISSUE THREAD

**Issue by grenzor** [keepassxreboot/keepassxc#1314]
I think it can be useful to be able to select multiple entries and edit. So some settings will be greyed out (the settings that can only edited individually) except for those that can be modified even when multiple entries are selected e.g. icons, additional attributes, enable/disable auto type settings and whatever else is applicable. This will be very useful and time saving instead of editing entries one by one.

**Comment by dnel** [comment]
I imported from Lastpass and would like to update the icon for over 500 entries which defaulted to the key. It's going to take a long time doing this without bulk update capabilities.

**Comment by rwese** [comment]
Since it is currently not possible for KeePassXC-Browser to have aliases I have for my corporate sites, all using activedirectory-login 20+ entries I need to update every 8 weeks.

So I would greatly appreciate this feature.

**Comment by shooftie** [comment]
When it comes to icons it might be nice to have a checkbox that just states "Inherit from group".

Additionally, it would be cool to be able to reference the group properties in the items via some kind of template tag e.g. "MyWebsite: { group_title }" would be displayed as "MyWebsite: HTTP Auth"

**Comment by nagromc** [comment]
> Since it is currently not possible for KeePassXC-Browser to have aliases I have for my corporate sites, all using activedirectory-login 20+ entries I need to update every 8 weeks.
> 
> So I would greatly appreciate this feature.

@rwese 
You can reference one central entry in other entries.
At work, I have one central ActiveDirectory entry, and every web app has its own entry pointing to.

Use `{REF:U@I:your-central-32-hex-digit-entry-ID}` in the _Username_ field, and `{REF:P@I:your-central-32-hex-digit-entry-ID}` in the _Password_ field.
For new entries, you can clone the central entry, and check _Replace username and password with references_.

**Comment by rwese** [comment]
Imho this is not fixed by #3273, it only covers icons but hardly the need for deeper bulk configuration changes, autotype, password, etc...

**Comment by droidmonkey** [comment]
Fair enough, this issue kind of expanded and contracted and expanded.

**Comment by grenzor** [comment]
#3273 doesn't really cover icons for this request either as that's per-group whereas icons meant under this request is for manually selecting multiple website entries then changing their icons in one go. Like changing icons of multiple entries in multiple different groups or changing icons of multiple entries in single big group with many different entries where the per-group option won't be useful.

**Comment by PrSunflower** [comment]
Hello 👋

I would also appreciate such a feature. I really like _KeePass_' "**Edit Entry (Quick)**" menu, I posted screenshots in #6527 🙂

**Comment by Vakke** [comment]
I would second this feature also. In desperate need to mass edit multiple login titles and have to use normal Keepass for that now.

**Comment by paivan1999** [comment]
The same to me, I would be very glad to use this feature.

**Comment by Tormen** [comment]
I had this idea but focused on Prefix + Suffix.
As this page does not mention "prefix" yet, please have a look at described UI proposal: https://github.com/keepassxreboot/keepassxc/issues/8070

```
feature detects if there is a common PREFIX and allows to EDIT it 
( if no common prefix is detected the field is empty 
  and a common Prefix can be added to all selected entries )
```

Thanks a lot in advance!

**Comment by hirak99** [comment]
Is there any command-line workaround?

E.g. say I want to add url to multiple logins. In the UI, I could select all of these and add a new tag "tobeedited". Is there any way I can use `keepassxc-cli` to iterate over a tag and set some field like the url?

Edit: Can we even add bulk tags from the UI? Just realized even that may not be possible 😞

**Comment by droidmonkey** [comment]
You can bulk add tags since 2.7.3

You can use the keepassxc cli to search for and edit entries in bulk.

**Comment by hfavisado** [comment]
As of 2.7.4, bulk adding of tags works well but only if none of the entries selected already have the tag to be added.
Bulk removing of tags works fine.

Suggestion: I think there should be a new UI containing the following behavior which exists in 
- a list of tags with checkboxes.
  - if only some of the selected items contain the tag, the checkbox is partially checked (light check mark?)
  - if ALL of the selected items contain the tag, the checkbox should be fully checked
  - if none of the selected items contain the tag, the checkbox should be unchecked
  - checking on a partially checked box makes it fully checked, that is, all the selected items will then have the tag after applying changes
- a search field at the top - this would be great if the list of tags is updated as text is written in the search field
- an OK button - saves the changes and closes the tag editing UI
- a cancel button - closes the tag editing UI and discards unsaved changes, if any exist
- KeepassXC doesn't follow material design but some guidelines in the "States" section of [this page](https://m2.material.io/components/checkboxes#behavior) might become good inspiration. I'm not so sure about using a box that looks like [-] for the indeterminate state.

**Comment by droidmonkey** [comment]
I explicitly made this functionality simple and will not be modifying it. If you want to add tags to all entries with mixed tags then you just need to right click through the tag menu twice.

**Comment by vkuznet** [comment]
@droidmonkey , I'm not sure I understand how to right click the tag menu as I do not see tag menu in macOS? Am I missing something, or is it platform specific? It would be extremely useful to have the following:
- ability to create/edit/manage tags separately, e.g. have a Edit tag menu option, etc.
- ability to select multiple records and choose tag we want to apply to them, instead of walking through individual records to fix them.
Can this be done on all platforms?

**Comment by droidmonkey** [comment]
You need to update to the latest version of keepassxc

**Comment by vkuznet** [comment]
what is exact version, I have the latest, 2.7.4 on macOS and I do not see this option.

**Comment by droidmonkey** [comment]
Right click any entry, you should see a Tags item. See this PR for more info: https://github.com/keepassxreboot/keepassxc/pull/8435

**Comment by zburgermeiszter** [comment]
It would be really useful to have the bulk update feature on password expiry date.
Is there any planned release date for 2.8.0?

**Comment by lvbibir** [comment]
How to chase icons for more than one entry at once, I want to change several entries to have the same icon.

**Comment by runil** [comment]
@lvbibir For this I found a simple solution that involves right clicking a parent folder which allows you to recursively apply an icon to all entries.

**Comment by t214c** [comment]
This seems like a popular issue. Any chance we see it soon-ish?

May I add that this should apply to groups too? Like batch change icon or searchability.

**Comment by git-developer** [comment]
> It would be really useful to have the bulk update feature on password expiry date. 

I agree that this would be really useful. This function is available as "Quick Edit" on original KeePass. See also https://github.com/keepassxreboot/keepassxc/issues/6527#issue-892509633 and https://github.com/keepassxreboot/keepassxc/discussions/7975 for other requests.

**Comment by nk-tedo-001** [comment]
Hey team!

I came to this issue to mention that it would be super useful, for example, to bulk-edit additional URLs for autocomplete.

**Comment by droidmonkey** [comment]
Related to this, and in fact might be preferred: #13661

**Comment by nk-tedo-001** [comment]
> Related to this, and in fact might be preferred: [#13661](https://github.com/keepassxreboot/keepassxc/pull/13661)

Do you mean preferred for merging or for use? Because in the case of use, I'm not sure.

Bulk editing is easy to handle - as a user, I can select multiple entries, enter edit mode, and set the same new value for some settings or delete custom attributes.

For example, we have around 20 test users that share a common login endpoint. Sometimes it changes, or I need to add a second URL. For now, it's a pain to manually edit each entry.

As I see it, the merge option could be useful when I need to update passwords or other settings with different values. In that case, I could import them as new entries and merge them into the existing ones.

**Comment by droidmonkey** [comment]
> For example, we have around 20 test users that share a common login endpoint. Sometimes it changes, or I need to add a second URL. For now, it's a pain to manually edit each entry.

You should use references. Make one entry the base one and then reference the username/password or whatever field you want. https://keepassxc.org/docs/KeePassXC_UserGuide#_entry_cross_reference

**Comment by nk-tedo-001** [comment]
> > For example, we have around 20 test users that share a common login endpoint. Sometimes it changes, or I need to add a second URL. For now, it's a pain to manually edit each entry.
> 
> You should use references. Make one entry the base one and then reference the username/password or whatever field you want. [keepassxc.org/docs/KeePassXC_UserGuide#_entry_cross_reference](https://keepassxc.org/docs/KeePassXC_UserGuide#_entry_cross_reference)

Unfortunately, it's blocked by https://github.com/keepassxreboot/keepassxc/issues/7380, as KeePassXC doesn't currently allow linking to `KP2A_URL` entries.

In my case, I have a main URL and several additional URLs per entry - being able to edit them in bulk would significantly improve the workflow.
